cmd/soroban-cli/src/commands/tx/args.rs

```diff
@@ -0,0 +1,107 @@
+use crate::{
+    commands::{global, txn_result::TxnEnvelopeResult},
+    config::{self, data, network, secret},
+    fee,
+    rpc::{self, Client, GetTransactionResponse},
+    tx::builder::{self, TxExt},
+    xdr::{self, Limits, WriteXdr},
+};
+
+#[derive(Debug, clap::Args, Clone)]
+#[group(skip)]
+pub struct Args {
+    #[clap(flatten)]
+    pub fee: fee::Args,
+    #[clap(flatten)]
+    pub config: config::Args,
+}
+
+#[derive(thiserror::Error, Debug)]
+pub enum Error {
+    #[error(transparent)]
+    Rpc(#[from] rpc::Error),
+    #[error(transparent)]
+    Config(#[from] config::Error),
+    #[error(transparent)]
+    Network(#[from] network::Error),
+    #[error(transparent)]
+    Secret(#[from] secret::Error),
+    #[error(transparent)]
+    Tx(#[from] builder::Error),
+    #[error(transparent)]
+    Data(#[from] data::Error),
+    #[error(transparent)]
+    Xdr(#[from] xdr::Error),
+}
+
+impl Args {
+    pub async fn tx(&self, body: impl Into<xdr::OperationBody>) -> Result<xdr::Transaction, Error> {
+        let source_account = self.source_account()?;
+        let seq_num = self
+            .config
+            .next_sequence_number(&source_account.to_string())
```

Comment:
We don't have to do this in this PR, but we should make sequence number an option, that if isn't set uses the nxt seq num lookup, otherwise uses the value passed in. A good follow up to add later.